crypto/signer.go

```diff
@@ -64,54 +53,39 @@ func NewSigner(sigAlgo sigpb.DigitallySigned_SignatureAlgorithm, signer crypto.S
 }
 
 // Sign obtains a signature after first hashing the input data.
-func (s Signer) Sign(data []byte) (sigpb.DigitallySigned, error) {
+func (s Signer) Sign(data []byte) (*sigpb.DigitallySigned, error) {
 	h := s.hash.New()
 	h.Write(data)
 	digest := h.Sum(nil)
 
 	if len(digest) != s.hash.Size() {
-		return sigpb.DigitallySigned{}, fmt.Errorf("hasher returned unexpected digest length: %d, %d",
+		return nil, fmt.Errorf("hasher returned unexpected digest length: %d, %d",
```

Comment:
Use the "got %d, want %d" format of error message to make it clear which of the two values is which.